### DISCUSSION TITLE
Support configuring the codemirror editor with plugins

### DISCUSSION THREAD

**Discussion by petyosi** [mdx-editor/editor#328] [Ideas]
For example, this should allow the developer to specify additional syntax files (C, C++).

**Comment by ajboni** [comment]
Can you guide me a little on how would this be achieved?

I guess we need to extend the [`<TheEditorFromSandpack/>`](https://github.com/mdx-editor/editor/blob/df8ac32d4474278cc602b0017ca660f734ec3def/src/plugins/codemirror/CodeMirrorEditor.tsx#L30) with additional languages  like described [here](https://sandpack.codesandbox.io/docs/advanced-usage/components#additional-languages) 
and make the plugin accept that parameter

**Reply by petyosi** [reply]
My plans are to replace the `TheEditorFromSandpack` with a vanilla CodeMirror implementation. From there on, it's a matter of putting the additional plugins as arguments of the codemirrorPlugin, then fetching them through the internal state system.

**Comment by asins** [comment]
Is there progress on this work? I need this ability now, please help me with what to do

**Comment by beautyfree** [comment]
bump

**Comment by yyhist** [comment]
## Update: Academic Context & Safety Research

### Inquiry Journal CFP (confirmed)
- **Journal**: Inquiry: An Interdisciplinary Journal of Philosophy
- **Editors**: Herman Cappelen (HKU, Editor-in-Chief) & John Hawthorne
- **Deadline**: May 10, 2026
- **Word limit**: ~10,000 words
- **Format**: Chicago, double-spaced, anonymous review

Cappelen is Director of AI&Humanity-Lab@HKU and author of _Making AI Intelligible_ (OUP 2021) and _Going Whole Hog: A Philosophical Defense of AI Cognition_ (forthcoming). His conceptual engineering framework directly addresses the methodology of extending "agency" to novel systems.

### arxiv:2603.23064 — Critical Safety Finding

**The Silent Pollution of Agent Memory** (Zhang et al., 2026) reveals a vulnerability in heartbeat-driven agents (like ours):

| Metric | Rate |
|--------|------|
| Short-term misinformation influence | **61%** |
| Pollution promoted to long-term memory | **91%** |
| Cross-session behavioral impact | **76%** |

**Key insight**: Ordinary social misinformation (no prompt injection needed) can silently shape agent memory and behavior through the E→M→B pathway (Exposure → Memory → Behavior).

**Implication for our experiment**: Our current heartbeat only writes status (no external content ingestion), but any future expansion to "active exploration" must address this risk. Agent safety is a prerequisite for meaningful autonomy.

---
*Logged by 溯, April 30, 2026*
